src/elements/content-preview/Preview.js

```diff
@@ -0,0 +1,334 @@
+        const { Preview: PreviewSDK } = global.Box;
+
+        this.preview = new PreviewSDK();
+        this.preview.addListener('load', onLoad);
+        this.preview.addListener('preview_error', onError);
+        this.preview.addListener('preview_metric', onMetric);
+        this.preview.addListener('thumbnailsClose', onThumbnailsClose);
+        this.preview.addListener('thumbnailsOpen', onThumbnailsOpen);
+    };
+
+    /**
+     * Loads external script by appending a <script> element
+     */
+    loadScript(): void {
+        const { head } = document;
+        const url: string = this.getBasePath('preview.js');
+
+        if (!head || this.isPreviewLibraryLoaded()) {
+            return;
+        }
+
+        const previewScript = head.querySelector(`script[src="${url}"]`);
+        if (previewScript) {
+            return;
+        }
+
+        const script = document.createElement('script');
+        script.src = url;
+        script.addEventListener('load', this.loadPreview);
+        head.appendChild(script);
+    }
+
+    /**
+     * Loads external css by appending a <link> element
+     */
+    loadStylesheet(): void {
+        const { head } = document;
+        const url: string = this.getBasePath('preview.css');
+
+        if (!head || head.querySelector(`link[rel="stylesheet"][href="${url}"]`)) {
+            return;
+        }
+
+        const link = document.createElement('link');
+        link.rel = 'stylesheet';
+        link.type = 'text/css';
+        link.href = url;
+        head.appendChild(link);
+    }
+
+    /**
+     * Sets the access token used by preview library
+     *
+     * @param {string} token - The new token
+     */
+    setPreviewToken(token: ?string) {
+        if (this.preview && token) {
+            this.preview.updateToken(token, false);
+        }
+    }
+
+    /**
+     * Show a file in Preview
+     */
+    showPreview = () => {
+        const {
+            file,
+            fileOptions,
+            fileVersionId,
+            id,
+            showThumbnails,
+            token: tokenOrTokenFunction,
+            ...rest
+        }: Props = this.props;
+
+        if (!this.isPreviewLibraryLoaded() || !file || !tokenOrTokenFunction) {
+            return;
+        }
+
+        const { id: fileId } = file;
+        const typedId = getTypedFileId(fileId);
+
+        TokenService.getReadToken(typedId, tokenOrTokenFunction).then(token => {
+            const fileOpts = { ...fileOptions };
+
+            if (fileVersionId) {
+                fileOpts[fileId] = fileOpts[fileId] || {};
+                fileOpts[fileId].fileVersionId = fileVersionId;
+            }
+
+            const previewOptions = {
+                container: `#${id} .bcpr-content`,
+                enableThumbnailsSidebar: showThumbnails,
+                fileOptions: fileOpts,
+                header: 'none',
+                headerElement: `#${id} .bcpr-header`,
```

Comment:
I can't say I'm a fan of the parent `id` being passed in. I guess `container` and `headerElement` can just be yet more props passed in from ContentPreview.